libr/core/rvc.c

```diff
@@ -733,30 +734,48 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
 		eprintf ("Can't commit");
 		return false;
 	}
-	if (!message) {
+	if (R_STR_ISEMPTY (message)) {
 		char *path = NULL;
 		(void)r_file_mkstemp ("rvc", &path);
 		if (path) {
+			char m[MAX_MESSAGE_LEN + 1];
 			char *editor = r_sys_getenv ("EDITOR");
```

Comment:
EDITOR is honored by cfg.editor. which is then used by r_cons_editor(), dont check for this env var. as its a unix-specific thing an r2 have the cfg.editor abstraction